cmd/ndm_daemonset/probe/eventhandler.go

```diff
@@ -63,32 +107,38 @@ func (pe *ProbeEvent) addBlockDeviceEvent(msg controller.EventMessage) {
 
 // deleteBlockDeviceEvent deactivate blockdevice resource using uuid from etcd
 func (pe *ProbeEvent) deleteBlockDeviceEvent(msg controller.EventMessage) {
-	blockDeviceOk := pe.deleteBlockDevice(msg)
-
-	// when one disk is removed from node and entry related to
-	// that disk is not present in etcd,  in that case it
-	// again rescan full system and update etcd accordingly.
-	if !blockDeviceOk {
-		go pe.initOrErrorEvent()
-	}
-}
-
-func (pe *ProbeEvent) deleteBlockDevice(msg controller.EventMessage) bool {
 	bdList, err := pe.Controller.ListBlockDeviceResource()
 	if err != nil {
 		klog.Error(err)
-		return false
 	}
-	ok := true
-	for _, diskDetails := range msg.Devices {
-		existingBlockDeviceResource := pe.Controller.GetExistingBlockDeviceResource(bdList, diskDetails.UUID)
+
```

Comment:
I'm not sure I understand what is happening here. It looks like, when we device gets deleted, it gets a new label?

Reply:
Nope. When a udev `remove` event occurs for a device we need to mark that BlockDevice as Inactive. So we will try to generate the UUID for the device which was removed. The label won't be recreated, we use the existing label and use it to identify the device which was removed.

Reply:
Ok, I see. If you remove the hash generation you can just delete the device directly.